sympy/core/tests/test_arit.py

```diff
@@ -378,14 +378,15 @@ def test_Add_Mul_is_integer():
 
 
 def test_Add_Mul_is_finite():
-    x = Symbol('x', real=True, finite=False)
+    x = Symbol('x', extended_real=True, finite=False)
 
     assert sin(x).is_finite is True
-    assert (x*sin(x)).is_finite is False
+    assert (x*sin(x)).is_finite is None
```

Comment:
Should `(x*sin(x)).is_finite` be False when x.is_finite is False?

I think of this as a nan type situation:
```
In [9]: (x*sin(x)).subs(x, oo)                                                                                                    
Out[9]: <-∞, ∞>

In [10]: print((x*sin(x)).subs(x, oo).is_zero)                                                                                    
None

In [11]: print((x*sin(x)).subs(x, oo).is_finite)                                                                                  
None
```